packages/schema/src/plugins/prisma/schema-generator.ts

```diff
@@ -124,10 +124,12 @@ export default class PrismaSchemaGenerator {
         }
         await writeFile(outFile, this.PRELUDE + prisma.toString());
 
+        const packageManager = process?.versions?.bun ? 'bunx' : 'npx';
+
         if (options.format === true) {
             try {
                 // run 'prisma format'
-                await execSync(`npx prisma format --schema ${outFile}`);
+                await execSync(`${packageManager} prisma format --schema ${outFile}`);
```

Comment:
Shall we extract an `execPackage` utility to encapsulate the pm detection here? I feel it's going to be needed in other places too in the future.

Reply:
Yeah, I thought about that but refrained from doing so because it was used only in that case in the schema package. However, I do agree and can implement this as it would likely ensure better support in the future. Do you have any recommendations about how you see this best done?

I notice there is already a [`getPackageManager` util function](https://github.com/zenstackhq/zenstack/blob/main/packages%2Fschema%2Fsrc%2Futils%2Fpkg-utils.ts) that I could export and adapt, but instead I think I'll create another function for package executors in specific, seeing that they do two very different things. I believe I'll put it in the same file, unless you think exec utils or another file would be a better place.

I'll leave tests using NPX for obvious reasons. I'll also leave other packages using NPX for now, considering I'm not familiar with the environment that they may be executed in.

Reply:
> Yeah, I thought about that but refrained from doing so because it was used only in that case in the schema package. However, I do agree and can implement this as it would likely ensure better support in the future. Do you have any recommendations about how you see this best done?
> 
> I notice there is already a [`getPackageManager` util function](https://github.com/zenstackhq/zenstack/blob/main/packages%2Fschema%2Fsrc%2Futils%2Fpkg-utils.ts) that I could export and adapt, but instead I think I'll create another function for package executors in specific, seeing that they do two very different things. I believe I'll put it in the same file, unless you think exec utils or another file would be a better place.
> 
> I'll leave tests using NPX for obvious reasons. I'll also leave other packages using NPX for now, considering I'm not familiar with the environment that they may be executed in.

Got it. Yes, npx should be the default as long as we're in a Node.js environment. I was just thinking of creating a new `exec...` utility (similar to the current `execSync` one) which encapsulates the bun detection (and npx as default) in it.

For the general task of package manager detection (npm, yarn, pnpm, etc.), I always wanted to migrate to using [@antfu/ni](https://www.npmjs.com/package/@antfu/ni), which Prisma also uses. But that should be a separate change.

Reply:
I've implemented the `execPackage` function in the latest commit which should do what you've outlined here. 

Also, it looks like [@antfu/ni](https://www.npmjs.com/package/@antfu/ni) supports getting a package executor (see the [nlx section](https://www.npmjs.com/package/@antfu/ni#nlx---download--execute)). I could work on another PR implementing this package for both the `getPackageManager` and `execPackage` functions after this one is merged.

Reply:
Got it. Thanks! I've approved this PR.

Just a reminder: I noticed that nlx translates to `pnpm dlx` and `yarn dlx`. IIRC, they always download the package to temp folder regardless if it's already installed in the current project. I think always resorting to "npx" should be safe for a Node environment?

Reply:
Cool! I'm merging this PR now. Is it blocking you now? The regular ship cadence is bi-weekly, so the 1.10.0 release will be at the end of Feb. If it's blocking, I can also publish a patch release for it.

Reply:
I have a workaround that I've been using for the past few months. There isn't any rush, it's just mainly to speed up build times and not copy files in and out of a docker container for one action.